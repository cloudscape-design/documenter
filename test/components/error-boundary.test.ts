// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { describe, expect, test } from 'vitest';
import { buildProject } from './test-helpers';

describe('Error boundary', () => {
  test('should have correct properties definitions', () => {
    const result = buildProject('error-boundary');
    expect(result).toMatchSnapshot();
  });
});
