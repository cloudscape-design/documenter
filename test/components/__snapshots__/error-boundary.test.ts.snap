// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error boundary > should have correct properties definitions 1`] = `
[
  {
    "dashCaseName": "error-boundary",
    "description": undefined,
    "events": [
      {
        "cancelable": false,
        "deprecatedTag": undefined,
        "description": "A callback that fires when the user clicks on the refresh button.",
        "detailInlineType": undefined,
        "detailType": undefined,
        "name": "onRefresh",
        "systemTags": undefined,
      },
    ],
    "functions": [],
    "name": "ErrorBoundary",
    "properties": [
      {
        "analyticsTag": undefined,
        "defaultValue": undefined,
        "deprecatedTag": undefined,
        "description": "A special callback that fires when an error is captured.",
        "i18nTag": undefined,
        "inlineType": {
          "name": "(error: Error, errorInfo: React.ErrorInfo) => void",
          "parameters": [
            {
              "name": "error",
              "type": "Error",
            },
            {
              "name": "errorInfo",
              "type": "React.ErrorInfo",
            },
          ],
          "returnType": "void",
          "type": "function",
        },
        "name": "onError",
        "optional": false,
        "systemTags": undefined,
        "type": "(error: Error, errorInfo: React.ErrorInfo) => void",
        "visualRefreshTag": undefined,
      },
    ],
    "regions": [
      {
        "deprecatedTag": undefined,
        "description": "React content.",
        "displayName": undefined,
        "i18nTag": undefined,
        "isDefault": true,
        "name": "children",
        "systemTags": undefined,
        "visualRefreshTag": undefined,
      },
    ],
    "releaseStatus": "stable",
    "systemTags": undefined,
  },
]
`;
